---
title: "Ottawa COVID-19 wastewater surveillance"
output: html_document
params:
  lang: "en"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
lang <- params$lang
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# libraries
options(scipen = 999)
library(plotly)
library(tidyverse)
library(jsonlite)
library(ggplot2)
library(dplyr)
library(tidyr)
library(lubridate)
library(stats)
library(zoo)
library(stringr)
library(EpiNow2)
library(data.table)
library(purrr)
library(RColorBrewer)

# scripts
source("../../R/observed_data.R")
source("../../R/wastewater.R")
source("../../R/epinow_functions.R")

# data
ott_observed_covid <- read.csv(file.path(getwd(), "../../Data/Observed data/OPH_Observed_COVID_Data.csv")) %>%
  mutate(date = as.Date(date))
ott_observed_waste <- read.csv("https://raw.githubusercontent.com/Big-Life-Lab/PHESD/main/Wastewater/Ottawa/Data/wastewater_virus.csv")
ott_long_ww_data <- read.csv("https://raw.githubusercontent.com/Big-Life-Lab/PHESD/main/Wastewater/Ottawa/Data/wwMeasure.csv")

ott_b117_data <- ott_observed_waste %>%
  mutate(sampleDate = as.Date(sampleDate)) %>%
  filter(detectB117 == TRUE)

# functions to integrate waste & covid data together
waste_clean <- wastewater_prep(ott_observed_waste)
ww_covid_waste <- waste_clean %>%
    filter(date >= "2020-06-02")

variant_data <- ww_long_prep(ott_long_ww_data) %>%
  select(date, aggregation, varB117) %>%
  filter(!is.na(varB117)) %>%
  spread(aggregation, varB117) %>%
  rename(propB117 = "single") %>%
  mutate(propB117 = propB117*100,
         sd = sd*100) %>%
  filter(!is.na(sd)) %>%
  mutate(propB117_7day = rollapply(propB117, width=7,
                  FUN=function(x) mean(x, na.rm = TRUE),
                  by=1, by.column=TRUE, partial=FALSE,
                  fill=NA, align="center")) %>%
  mutate(sd_7day = rollapply(sd, width=7,
                  FUN=function(x) mean(x, na.rm = TRUE),
                  by=1, by.column=TRUE, partial=FALSE,
                  fill=NA, align="center"))

variant_non_detect <- ww_long_prep(ott_long_ww_data) %>%
  select(date, aggregation, varB117) %>%
  filter(!is.na(varB117)) %>%
  spread(aggregation, varB117) %>%
  rename(propB117_detect = "single") %>%
  select(date, propB117_detect) %>%
  filter(propB117_detect == 0) %>%
  full_join(variant_data, by = "date")

# Colours
case_col <- "rgb(0, 128, 128)"
case_shade <- "rgba(0, 128, 128, 0.2)"
var_col <- "rgb(85, 151, 217)"
n1_col <- "rgb(23, 63, 95)"
n2_col <- "rgb(60, 174, 163)"
n1_n2_col <- "rgb(237, 85, 59)"
rolling_avg_col <- "rgb(226, 127, 88)"
smooth_col <- "rgb(34, 140, 34)"
red <- "rgb(255,0,0)"
red_shade <- "rgba(255, 0, 0, 0.2)"
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Initialize calls for each parameter
new_case_call <- list(type = "observed_data", y_column = "observed_new_cases", name = "Daily reported new cases", short_name = "Daily cases", color = case_col)

new_case_5_day_call <- list(type = "observed_data", y_column = "observed_new_cases_5_day", name = "5-day rolling mean in \nreported new cases \n(by reporting date)", short_name = "New cases\n(reported date)", color = case_col, opacity = 0.15)

new_case_10_day_call <- list(type = "observed_data", y_column = "observed_new_cases_10_day", name = "10-day rolling mean in \nreported new cases \n(by reporting date)", short_name = "New cases\n(reported date)", color = case_col, opacity = 0.15)

new_case_7_day_call <- list(type = "observed_data", y_column = "observed_new_cases_7_day", name = "7-day rolling mean in \nreported new cases \n(by reporting date)", short_name = "New cases\n(reported date)", color = case_col, opacity = 0.15)

new_episode_5_day_call <- list(type = "observed_data", y_column = "observed_new_episodes_5_day", name = "5-day mean in \nreported new cases \n(by episode date)", short_name = "New cases\n(episode date)", color = case_col, opacity = 0.15)

new_episode_10_day_call <- list(type = "observed_data", y_column = "observed_new_episodes_10_day", name = "10-day mean in \nreported new cases \n(by episode date)", short_name = "New cases\n(episode date)", color = case_col, opacity = 0.15)

new_episode_7_day_call <- list(type = "observed_data", y_column = "observed_new_episodes_7_day", name = "7-day mean in \nreported new cases \n(by episode date)", short_name = "New cases\n(episode date)", color = case_col, opacity = 0.15)

new_community_case_5_day_call <- list(type = "observed_data", y_column = "observed_new_community_cases_5_day", name = "5-day mean \nin reported new community cases", color = case_col, opacity = 0.15)

new_community_case_10_day_call <- list(type = "observed_data", y_column = "observed_new_community_cases_7_day", name = "10-day mean \nin reported new community cases", color = case_col, opacity = 0.15)

new_community_case_7_day_call <- list(type = "observed_data", y_column = "observed_new_community_cases_10_day", name = "7 day mean \nin reported new community cases", color = case_col, opacity = 0.15)

hosp_call <- list(type = "observed_data", y_column = "observed_census_ICU_p_acute_care", name = "Hospital count", short_name = "Hosp count", color = case_col, opacity = 0.15)

pct_positivity_call_7_day <- list(type = "observed_data", y_column = "pct_positivity_7_day", name = "7-day mean \npercent positivity rate", short_name = "% positivity", color = case_col, opacity = 0.15)

active_case_call <- list(type = "observed_data", y_column = "observed_active_cases", name = "Active cases", short_name = "Active cases", color = case_col, opacity = 0.15)

active_case_5_day_call <- list(type = "observed_data", y_column = "observed_active_cases_5_day", name = "5-day mean \nin active cases", color = case_col, opacity = 0.15)

active_case_7_day_call <- list(type = "observed_data", y_column = "observed_active_cases_7_day", name = "7-day mean \nin active cases", short_name = "Active cases", color = case_col, opacity = 0.15)

active_case_10_day_call <- list(type = "observed_data", y_column = "observed_active_cases_10_day", name = "10-day mean \nin active cases", color = case_col, opacity = 0.15)

n1_call <- list(type = "signal_data", y_column = "N1", name = "N1 RNA signal", color = n1_col, yaxis = "y2") 

n2_call <- list(type = "signal_data", y_column = "N2", name = "N2 RNA signal", color = n2_col, yaxis = "y2")

n1_n2_call <- list(type = "avg_data", y_column = "N1_N2_avg", name = "Avg. viral signal", short_name = "viral signal", color = n1_n2_col, yaxis = "y2", opacity = 0.15, width = 4, showlegend = TRUE)

n1_n2_10_day_call <- list(type = "avg_data", y_column = "N1_N2_10_day", name = "10-day rolling mean viral signal", short_name = "10-day", color = rolling_avg_col, yaxis = "y2", width = 4, showlegend = TRUE)

n1_n2_5_day_call <- list(type = "avg_data", y_column = "N1_N2_5_day", name = "5-day rolling mean viral signal", short_name = "5-day", color = rolling_avg_col, yaxis = "y2", width = 4, showlegend = TRUE)

n1_n2_7_day_call <- list(type = "avg_data", y_column = "N1_N2_7_day", name = "7-day rolling mean viral signal", short_name = "7-day", color = rolling_avg_col, yaxis = "y2", width = 4, showlegend = TRUE)

infa_7_day <- list(type = "avg_data", y_column = "InfA_7_day", name = "7-day rolling mean Influenza A viral signal", short_name = "7-day", color = rolling_avg_col, yaxis = "y2", width = 4, showlegend = TRUE)

rsv_7_day <- list(type = "avg_data", y_column = "RSV_7_day", name = "7-day rolling mean RSV viral signal", short_name = "7-day", color = rolling_avg_col, yaxis = "y2", width = 4, showlegend = TRUE)

viral_roc_call <- list(type = "avg_data", y_column = "viral_roc_daily", name = "Daily rate of change in viral signal", color = rolling_avg_col, yaxis = "y2", width = 4, showlegend = TRUE)

rolling_avg <- list(type = "avg_data", y_column = "rolling_avg", name = "Daily rate of change in viral signal", color = rolling_avg_col, yaxis = "y2", width = 4, showlegend = TRUE)

viral_rolling_avg_call <- list(type = "avg_data", y_column = "avg_viral_roc_5_day", name = "5-day rolling avg. daily \nchange in viral signal", color = rolling_avg_col, yaxis = "y2", width = 4, showlegend = TRUE)

change_N1_N2_5_day_call <- list(type = "avg_data", y_column = "change_N1_N2_5_day", name = "Percent change in rolling avg \nover 5 days", short_name = "5-day % change", color = rolling_avg_col, yaxis = "y2", width = 4, showlegend = TRUE)

change_N1_N2_10_day_call <- list(type = "avg_data", y_column = "change_N1_N2_10_day", name = "Percent change in rolling avg \nover 10 days", short_name = "10-day % change", color = rolling_avg_col, yaxis = "y2", width = 4, showlegend = TRUE)

change_N1_N2_7_day_call <- list(type = "avg_data", y_column = "change_N1_N2_7_day", name = "Percent change in \nviral signal rolling \navg over 7 days", short_name = "7 day % change", color = rolling_avg_col, yaxis = "y2", width = 4, showlegend = TRUE)

change_new_cases_5_day_call <- list(type = "observed_data", y_column = "change_new_cases_5_day", name = "Percent change in rolling avg \nover 5 days", short_name = "5 day % change", color = case_col, yaxis = "y2", opacity = 0.15)

change_new_cases_10_day_call <- list(type = "observed_data", y_column = "change_new_cases_10_day", name = "Percent change in rolling avg \nover 10 days", short_name = "10 day % change", color = case_col, yaxis = "y2", opacity = 0.15)

change_new_cases_7_day_call <- list(type = "observed_data", y_column = "change_new_cases_7_day", name = "Percent change in \nnew case rolling \navg over 7 days", short_name = "7 day % change", color = case_col, yaxis = "y2", opacity = 0.15)

daily_infa_signal <- list(type = "observed_data", y_column = "InfA", name = "Daily Influenza A viral signal", short_name = "Daily signal", color = rolling_avg_col, yaxis = "y2", opacity = 0.50)

daily_rsv_signal <- list(type = "observed_data", y_column = "RSV", name = "Daily RSV viral signal", short_name = "Daily signal", color = rolling_avg_col, yaxis = "y2", opacity = 0.50)

daily_viral_signal_call <- list(type = "observed_data", y_column = "N1_N2_avg_clean", name = "Daily viral signal", short_name = "Daily signal", color = rolling_avg_col, yaxis = "y2", opacity = 0.50)

daily_viral_signal_call_omit <- list(type = "observed_data", y_column = "N1_N2_avg_omit", name = "Omitted viral signal", short_name = "Daily signal", color = case_col, yaxis = "y2", opacity = 0.50)

var_call <- list(type = "signal_data", y_column = "propB117", name = "% B.1.1.7 RNA", color = case_col)

var_nondetect_call <- list(type = "signal_data", y_column = "propB117_detect", name = "B.1.1.7 non-detection", color = red)

var_7_day_avg <- list(type = "avg_data", y_column = "propB117_7day", name = "7 observation \nrolling average", color = var_col, width = 3)
```

```{r include=FALSE}
##Load localization file
source("../../R/localization.R")
localization_data <- read.csv(file.path(getwd(),"../../Data/translations/Covid Localization - Home Page.csv"))


```

```{r, results='hold', echo=FALSE, comment=NA}
time <- .POSIXct(Sys.time(), "America/New_York")
wwintro<- insert_translation(localization_data, lang, "WWIntro", c(format(time, '%d %B, %Y, %X')))
end_transition_out <- insert_translation(localization_data, lang, "end_transition_out", c(format(time, '%d %B, %Y, %X')))
curr_day <- as.Date(time)
collect_date <- as.Date(last(ww_covid_waste$date))
wwcollectdate<- insert_translation(localization_data, lang, "WWCollectDate", c(format(collect_date, '%d %B, %Y')))
wwanalysesdate<- insert_translation(localization_data, lang, "WWAnalysesDate", c(format(curr_day, '%d %B, %Y')))
```

`r end_transition_out`

`r wwintro`

`r wwcollectdate`

`r wwanalysesdate`

The website is updated Monday to Friday at approximately 5 p.m. Updates can be delayed if measurements do not pass quality assurance and quality control checks.

Samples that are collected from Monday to Thursday are processed the day after collection. The website and plots are updated the day after processing. Samples that are collected from Friday to Sunday are processed on Monday and thus are posted to the website on Tuesday at 5 p.m.

People with COVID-19 shed the causative SARS-CoV-2 virus in their stool, regardless of whether they have symptoms, receive a COVID-19 test or ever are diagnosed. Thus, in contrast to assessing community COVID-19 levels by measuring the number of active cases, which may miss asymptomatic infections as well as be subject to limited test availability, wastewater surveillance consistently captures most of the population with COVID-19 given that everyone goes to the washroom. In addition to serving as a valuable confirmatory data source for COVID-19 levels, wastewater can also serve as early indicator for possible outbreaks, as described below.

------------------------------------------------------------------------

#  {#ww-visualization}

For mobile users, dragging the plot while in landscape mode will allow you to view current data. Using the display options at the top allows you to modify the view by zooming in and out of the plot.

Data last reported `r as.character(tail(waste_clean$reportDate, n=1))`. Click [here](/model/#wastewater) for more details on wastewater data reporting.

```{r, echo=FALSE, warning=FALSE, message=FALSE}
new_ott_covid_waste <- copy(ww_covid_waste)
temp_data <- new_ott_covid_waste %>% subset(!is.na(N1_N2_avg_omit))
end_data_first <- first(temp_data$date)
start_data_second <- last(temp_data$date)
new_ott_covid_waste$N1_N2_7_day_p1 <- NA
new_ott_covid_waste[(new_ott_covid_waste$date < end_data_first),"N1_N2_7_day_p1"] <- new_ott_covid_waste[(new_ott_covid_waste$date < end_data_first), "N1_N2_7_day"]
new_ott_covid_waste$N1_N2_7_day_p2 <- NA
new_ott_covid_waste[(new_ott_covid_waste$date > start_data_second),"N1_N2_7_day_p2"] <- new_ott_covid_waste[(new_ott_covid_waste$date > start_data_second), "N1_N2_7_day"]

n1_n2_7_day_call_p1 <- list(type = "avg_data", y_column = "N1_N2_7_day_p1", name = "7-day rolling mean viral signal", short_name = "7-day", color = rolling_avg_col, yaxis = "y2", width = 4, showlegend = FALSE)

n1_n2_7_day_call_p2 <- list(type = "avg_data", y_column = "N1_N2_7_day_p2", name = "7-day rolling mean viral signal", short_name = "7-day", color = rolling_avg_col, yaxis = "y2", width = 4, showlegend = TRUE)

reworked_figure(xaxis = "date", yaxis = list(daily_viral_signal_call, daily_viral_signal_call_omit, n1_n2_7_day_call_p1, n1_n2_7_day_call_p2), titles = c(y = "Normalized viral copies*", y2 = "", x = "Date", title = "<b>COVID-19 wastewater viral signal, Ottawa</b>"), data = new_ott_covid_waste, date_constraint = TRUE, ticks = FALSE, constraint_val = 40, ticklabels = "%b %d")

```

\*Rolling mean viral signal omits dates (coloured in blue) flagged by wastewater researchers with potential data concerns. Data is currently being studied for effects of snow melt and other diluting factors on RNA signal.

------------------------------------------------------------------------

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.asp = 0.75}
#reworked_figure(xaxis = "date", yaxis = list(var_nondetect_call, var_call, var_7_day_avg), error_bands = TRUE, error_pct = TRUE, error_data = "sd_7day", error_col = case_shade, titles = c(y = "Proportion (%)", y2 = "", x = "Date", title = "<b>Proportion of variant B.1.1.7 RNA signal in wastewater, Ottawa</b>"), data = variant_non_detect, ticks = FALSE)
```

------------------------------------------------------------------------

<!-- ## Wastewater Projections

Preliminary projections forecasting viral signal are shown below. Projections forecast normalized viral copies per copy based on observed signal data. Caution is encouraged when interpreting results. Methods have not been peer-reviewed, and validation is ongoing. -->

```{r, echo=FALSE, warning=FALSE, message=FALSE}
source("../../R/epinow_functions.R")
load("../../Data/short_term_ww_proj.RData")

ww_data <-
  read.csv("https://raw.githubusercontent.com/Big-Life-Lab/PHESD/main/Wastewater/Ottawa/Data/wastewater_virus.csv")

ww_clean <- wastewater_prep(ww_data) %>%
  select(date, N1_N2_avg, N1_N2_avg_clean) %>%
  mutate(date = as.Date(date))

ww_normalized <- ww_clean %>%
  mutate(N1_N2_avg_clean = N1_N2_avg_clean*1000000)

#short_term_plot(
#  projections = ww_forecast[[1]],
#  obs_data = ww_normalized,
#  obs_column = "N1_N2_avg_clean",
#  forecast_type = "reported_cases",
#  ylab = "Normalized viral copies* (µ)",
#  title = "<b>Projections for COVID-19 wastewater viral signal, Ottawa</b>",
#  tick_period = "1 month",
#  tick_labels_date = "%b %y",
#  annotation_text = "*Shaded area represents the 75% credible region",
#  interval_num = 178
#)
```

------------------------------------------------------------------------

## Current growth estimates for the amount of virus in wastewater

```{r, echo=FALSE, warning=FALSE, message=FALSE}
growth_measures <- ww_forecast[[2]]
colnames(growth_measures) <- c("measure", "estimate")

growth_measures[1, "measure"] <- "Expected change in viral amount"
growth_measures <- growth_measures[-2,]

knitr::kable(growth_measures)
```

------------------------------------------------------------------------

```{r, echo=FALSE, warning=FALSE, message=FALSE}
source("../../R/variant_df_prep.R")
library(readxl)

variant_sheet <- as.data.frame(read_excel(file.path(getwd(), "../../Data/variant_sheet.xlsx")))

names(variant_sheet) <- c("ENT_ID", "date_collected", "date_received_nml", "report_date", "turnaround_days","assay_id", "CDC_N1", "CDC_N2", "breadth_of_coverage_pct","mean_coverage_depth", "pangolin_lineage", "detect","prop_voc_mutations", "detection_quality", "frac_voc_detected")

sheet <- variant_sheet %>%
  select("ENT_ID", "date_collected", "date_received_nml", "report_date", "turnaround_days",
                         "assay_id", "CDC_N1", "CDC_N2", "breadth_of_coverage_pct",
                         "mean_coverage_depth", "pangolin_lineage",
                         "prop_voc_mutations", "detection_quality", "frac_voc_detected") %>%
  filter(!is.na(ENT_ID))

#variant_text <- variant_df_prep(sheet)

#date <- format(as.Date(variant_text[variant_text$variable == "date",2]), "%B %d, %Y")

#alphaDetect <- as.logical(variant_text[variant_text$variable == "alphaDetect",2])

#alphaLevel <- as.character(variant_text[variant_text$variable == "alphaLevel",2])

#alphaSummary <- ifelse(alphaDetect == TRUE, paste("Alpha (B.1.1.7) detected in", alphaLevel, "levels by sequencing and PCR, last reported on", paste0(date, ".")), paste("Alpha (B.1.1.7) not detected, last reported on", paste0(date, ".")))

#deltaDetect <- as.logical(variant_text[variant_text$variable == "deltaDetect",2])

#deltaLevel <- as.character(variant_text[variant_text$variable == "deltaLevel",2])

#deltaSummary <- ifelse(deltaDetect == TRUE, paste("B.1.617+ including Delta variant detected at", deltaLevel, "levels by sequencing, last reported on", paste0(date, ".")), paste("B.1.617+ including Delta variant not detected by sequencing, last reported on", paste0(date, ".")))

#b1351Detect <- as.logical(variant_text[variant_text$variable == "B1351Detect",2])

#b1351Level <- as.character(variant_text[variant_text$variable == "B1351Level",2])

#b1351Summary <- ifelse(b1351Detect == TRUE, paste("B.1.351 detected at", b1351Level, "levels by sequencing, last reported on", paste0(date, ".")), paste("B.1.351 not detected by sequencing, last reported on", paste0(date, ".")))

#p1Detect <- as.logical(variant_text[variant_text$variable == "P1Detect",2])

#p1Level <- as.character(variant_text[variant_text$variable == "P1Level",2])

#p1Summary <- ifelse(deltaDetect == TRUE, paste("P.1 variant detected at", deltaLevel, "levels by sequencing, last reported on", paste0(date, ".")), paste("P.1 variant not detected by sequencing, last reported on", paste0(date, ".")))

#otherVOCDetect <- as.logical(variant_text[variant_text$variable == "otherVOCDetect",2])

#otherVOCLevel <- as.character(variant_text[variant_text$variable == "otherVOCDetails",2])

#otherVOCSummary <- ifelse(otherVOCDetect == TRUE, otherVOCDetails, "No other VOC/VOI were identified by sequencing above the level of detection of the assay.")

#summaryDescription <- as.character(variant_text[variant_text$variable == "summaryDescription",2])
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#detailedDescription1 <- as.character(variant_text[variant_text$variable == "detailedDescription1",2])

#detailedDescription2 <- as.character(variant_text[variant_text$variable == "detailedDescription2",2])

#detailedDescription3 <- as.character(variant_text[variant_text$variable == "detailedDescription3",2])

#detailedDescription4 <- as.character(variant_text[variant_text$variable == "detailedDescription4",2])
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#ctN1Assay <- variant_text[variant_text$variable == "ctN1Assay",2]

#ctN2Assay <- variant_text[variant_text$variable == "ctN2Assay",2]
```

<!-- ### Variants of Concern --> 

<!-- removing the section on VOCs and the accompanying figure, but leaving this placeholder in anticipation of some potential text inputs from Tyson--> 

------------------------------------------------------------------------

### Overall viral signal

-   Standard curves R 2 ≥ 0.95.

-   Primer efficiency between 90%-130%.

-   No template controls are negative.

------------------------------------------------------------------------

### Influenza virus

Samples are screened for influenza A virus (IAV) once a week during the summer period (April to September). Samples are screened daily for IAV and influenza B virus (IBV) during the winter period (October to March). The website and plots are updated the day after processing.

The next plot illustrate the 7-day rolling mean in daily IAV wastewater viral signal. This number is the average of a week's daily readings.

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Create subset of data with only needed columns. Subset is needed so only valid columns are checked for NA, other columns would cause unnecessary trimming.
inf_data <- ww_covid_waste[,c( "date", "InfA", "InfA_7_day", "RSV", "RSV_7_day")]

# Trim any rows with NA in any column, only affecting trailing rows
inf_data <- na.trim(inf_data,sides="right")


p <- reworked_figure(xaxis = "date", yaxis = list(daily_infa_signal, infa_7_day), titles = c(y = "Normalized Influenza A viral copies*", y2 = "", x = "Date", title = "<b>Influenza A wastewater viral signal, Ottawa</b>"), data = inf_data, date_constraint = TRUE, ticks = FALSE, constraint_val = 40, ticklabels = "%b %d")

p <- layout(p, yaxis = list(range = c(0, 0.0001350000), exponentformat="none", tickformat=".5f"))
p
```

### Respiratory Syncytial Virus Infection (RSV)

Samples are screened daily for respiratory syncytial virus (RSV) during the winter period (October to March). This test does not differentiate between RSV A and RSV B strains. The website and plots are updated the day after processing.

The next plots illustrate the 7-day rolling mean in daily RSV wastewater viral signal. This number is the average of a week's daily readings. RSV is not a reportable disease, meaning that it does not fall under the list of diseases of public health significance in Ontario as designated by the Minister of Health$^1$. Nevertheless, RSV has been identified as a disease that could benefit from wastewater-based surveillance given its contribution to seasonal emergency department visits and hospitalizations among young children and to outbreaks in healthcare institutions such as long-term care homes. Visit [Respiratory Virus Detections](https://www.canada.ca/en/public-health/services/surveillance/respiratory-virus-detections-canada.html) in Canada for more information on RSV surveillance.

```{r, echo=FALSE, warning=FALSE, message=FALSE}
p <- reworked_figure(xaxis = "date", yaxis = list(daily_rsv_signal, rsv_7_day), titles = c(y = "Normalized RSV viral copies*", y2 = "", x = "Date", title = "<b>RSV wastewater viral signal, Ottawa</b>"), data = inf_data, date_constraint = TRUE, ticks = FALSE, constraint_val = 40, ticklabels = "%b %d")
p <- layout(p, yaxis = list(range = c(0, 0.0008500), exponentformat="none", tickformat=".5f"))
p
```

------------------------------------------------------------------------

## Interpretation cautions

The accuracy and reliability of SARS-CoV-2, influenza and RSV wastewater testing is improving as scientists understand the role of factors such as differences in sewage systems and laboratory protocols. Nonetheless, we recommend caution when interpreting daily or less then 3-day variation in the viral signal while evaluation of the methodology in public health surveillance is ongoing. Research is underway to arrive at a more precise estimate of sensitivity. For influenza and RSV, the level of detection is currently unknown as less than a year's worth of data has been collected, but evaluation is ongoing. Thus, direct comparisons between wastewater signal levels for the 3 viruses is not recommended at this time. COVID-19, influenza and RSV wastewater signal is helpful when interpreted alongside other surveillance measures, taking into consideration the strengths and limitations of each measure. It is most promising as an early indicator$^2$ of community viral activity and illness and may serve as a warning in advance of increased care-seeking, outbreaks and hospitalizations.

Visit the Ottawa Public Health [Respiratory and Enteric Surveillance Report](https://www.ottawapublichealth.ca/en/reports-research-and-statistics/flu-report.aspx) for more information on Ottawa's public health surveillance of respiratory and enteric viruses.

For more information on wastewater testing in Ottawa, please see the [contact](https://613covid.ca/contact/) page.

Citations:

1.  Health Protection and Promotion Act, R.S.O. 1990, c. H.7. <https://www.ontario.ca/laws/statute/90h07?search=diseases+of+public+health+significance>.

2.  Mercier, É. et al. Wastewater surveillance of influenza activity: Early detection, surveillance, and subtyping in city and neighbourhood communities (preprint). (2022) <doi:10.1101/2022.06.28.22276884>.

------------------------------------------------------------------------

See the [Methods](/model/) page for more information on how the samples were collected, access to the data, and how the plots were created. The plots are currently for research only and presented to the public for discussion.

You can learn more about wastewater epidemiology and its role in COVID-19 surveillance on Ottawa Public Health's [website](https://www.ottawapublichealth.ca/en/reports-research-and-statistics/Wastewater_COVID-19_Surveillance.aspx).

## Definitions

\* A 7-day average is generated by averaging the levels from a given day with the six previous days. The average is termed "rolling" as it changes each day.

\* For new cases, the reported date is the day the test result is reported by the laboratory. Episode date is the approximate date of COVID-19 infection estimated from information available: the date of symptom onset, test date, or the reported date.

\* Percent positivity refers to the percentage of individuals tested for either COVID-19 infection or influenza A infection in a given day that test positive.

\* A central question in wastewater epidemiology is determining the proportion of the wastewater that is actually from humans and the proportion that is rain water, snow melt etc. To address this issue, viral copy data is thus normalized using a seasonally stable fecal biomarker; pepper mild mottle virus (PMMoV). See [methods](/model/) for more details.

\^ Percent change in 7-day average is calculated by comparing the 7-day average (previous day 1 to 7) with a lagged 7-day average (days 8 to 14).
